From 2f2145f46b9ff2493428bbfbb48a0eb52fb34e94 Mon Sep 17 00:00:00 2001
From: zhongfly <11155705+zhongfly@users.noreply.github.com>
Date: Wed, 29 Jul 2026 21:56:22 +0800
Subject: [PATCH] libsrt: disable pthread_atfork

MinGW exposes the symbol during CMake checks but does not declare it for the libsrt build, causing api.cpp to fail.
---
 packages/libsrt.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/libsrt.cmake b/packages/libsrt.cmake
index 78a64d3..28810b4 100644
--- a/packages/libsrt.cmake
+++ b/packages/libsrt.cmake
@@ -15,6 +15,7 @@ ExternalProject_Add(libsrt
         -DENABLE_STDCXX_SYNC=ON
         -DENABLE_APPS=OFF
         -DENABLE_SHARED=OFF
+        -DHAVE_PTHREAD_ATFORK=OFF
         -DUSE_ENCLIB=openssl
         -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     BUILD_COMMAND ${EXEC} ninja -C <BINARY_DIR>
-- 
2.54.0.windows.1

